app/component/itinerary/NaviStack.js

```diff
@@ -29,10 +39,12 @@ NaviStack.propTypes = {
   ).isRequired,
   handleRemove: PropTypes.func.isRequired,
   cardExpanded: PropTypes.bool,
+  legType: PropTypes.string,
 };
 
 NaviStack.defaultProps = {
   cardExpanded: false,
+  legType: '',
```

Comment:
legType does not have logical connection to NaviStack which may contain messages for many legs. I suggest passing the class postfix itself, or maybe handling the vertical offset completely in parent component.